doctr/models/recognition/master/pytorch.py

```diff
@@ -0,0 +1,290 @@
+            *conv_sequence_pt(512, 512, relu=True, bn=True, kernel_size=3, padding=1),
+        ]
+        super().__init__(*_layers)
+
+
+class MASTER(_MASTER, nn.Module):
+
+    """Implements MASTER as described in paper: <https://arxiv.org/pdf/1910.02562.pdf>`_.
+    Implementation based on the official Pytorch implementation: <https://github.com/wenwenyu/MASTER-pytorch>`_.
+
+    Args:
+        vocab: vocabulary, (without EOS, SOS, PAD)
+        d_model: d parameter for the transformer decoder
+        headers: headers for the MAGC module
+        dff: depth of the pointwise feed-forward layer
+        num_heads: number of heads for the mutli-head attention module
+        num_layers: number of decoder layers to stack
+        max_length: maximum length of character sequence handled by the model
+        input_size: size of the image inputs
+    """
+
+    def __init__(
+        self,
+        vocab: str,
+        d_model: int = 512,
+        headers: int = 1,
+        dff: int = 2048,
+        num_heads: int = 8,
+        num_layers: int = 3,
+        max_length: int = 50,
+        input_shape: Tuple[int, int, int] = (3, 48, 160),
+        cfg: Optional[Dict[str, Any]] = None,
+    ) -> None:
+        super().__init__()
+
+        self.max_length = max_length
+        self.vocab = vocab
+        self.cfg = cfg
+        self.vocab_size = len(vocab)
+        self.num_heads = num_heads
+
+        self.feature_extractor = MAGCResnet(headers=headers)
+        self.seq_embedding = nn.Embedding(self.vocab_size + 1, d_model)  # One additional class for EOS
+
+        self.decoder = Decoder(
+            num_layers=num_layers,
+            d_model=d_model,
+            num_heads=num_heads,
+            dff=dff,
+            vocab_size=self.vocab_size,
+            maximum_position_encoding=max_length,
+        )
+        self.feature_pe = positional_encoding(input_shape[1] * input_shape[2], d_model)
+        self.linear = nn.Linear(d_model, self.vocab_size + 1)
+        nn.init.kaiming_uniform_(self.linear.weight)
+
+        for m in self.modules():
+            if isinstance(m, nn.Conv2d):
+                nn.init.kaiming_normal_(m.weight, mode='fan_out', nonlinearity='relu')
+            elif isinstance(m, (nn.BatchNorm2d, nn.GroupNorm)):
+                nn.init.constant_(m.weight, 1)
+                nn.init.constant_(m.bias, 0)
+
+    def make_mask(self, target: torch.Tensor) -> torch.Tensor:
+        look_ahead_mask = create_look_ahead_mask(target.size(1))
+        target_padding_mask = create_padding_mask(target, self.vocab_size)  # Pad with EOS
+        combined_mask = target_padding_mask & look_ahead_mask
+        return torch.tile(combined_mask.permute(1, 0, 2), (self.num_heads, 1, 1))
+
+    def forward(
+        self,
+        x: torch.Tensor,
+        target: Optional[List[str]] = None,
+        return_model_output: bool = False,
+        return_preds: bool = False,
+        **kwargs: Any,
+    ) -> torch.Tensor:
```

Comment:
would you mind adopting the same output type as other recognition models please ? :pray: (dict with the same keys)